sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/display/DisplayData.java

```diff
@@ -883,7 +883,7 @@ private DisplayData build() {
       }
       Type type = inferType(got);
       if (type == null) {
-        throw new RuntimeException(String.format("Unknown value type: %s", got));
+        return item(key, Type.STRING, got.toString());
```

Comment:
I'm not sure if we want to always assume string type when we fail to infer the type

Reply:
We already assume string type if it's not accessible (we return a `STRING` containing the whole provider's `toString`), so I think it's not too big of a stretch to also do that when we can't infer the type of the stored value.

Main reason I did this change is that I wanted this function to work fine for things like `ValueProvider<ResourceId>` without the user needing to do the `isAccessible` check and doing a `toString` themselves anyway in all cases.